pkg/db/canary.go

```diff
@@ -77,9 +77,13 @@ func PersistCheck(check pkg.Check) (string, error) {
 
 func DeleteCanary(canary v1.Canary) error {
 	logger.Infof("deleting canary %s/%s", canary.Namespace, canary.Name)
-	model := pkg.CanaryFromV1(canary)
+	model, err := pkg.CanaryFromV1(canary)
+	if err != nil {
+		return err
+	}
 	deleteTime := time.Now().Time
-	tx := Gorm.Find(&model).Where("id = ?", *canary.Status.PersistedID).UpdateColumn("deleted_at", deleteTime)
+	persistedID := canary.GetPersistedID()
```

Comment:
Do I also need to check and error if this is "" ?

Reply:
If persistedID is nil or "" then we can skip the db delete